scripts/imgtool/image.py

```diff
@@ -81,21 +94,30 @@ def load(cls, path, included_header=False, **kwargs):
         obj.check()
         return obj
 
-    def __init__(self, version=None, header_size=IMAGE_HEADER_SIZE, pad=0):
+    def __init__(self, version=None, header_size=IMAGE_HEADER_SIZE, pad=0,
+                 hex_base_addr=None):
         self.version = version or versmod.decode_version("0")
         self.header_size = header_size or IMAGE_HEADER_SIZE
         self.pad = pad
+        self.hex_base_addr = hex_base_addr
 
     def __repr__(self):
-        return "<Image version={}, header_size={}, pad={}, payloadlen=0x{:x}>".format(
+        return "<Image version={}, header_size={}, base_addr={}, pad={}, \
+                payloadlen=0x{:x}>".format(
                 self.version,
                 self.header_size,
+                self.hex_base_addr or "N/A",
                 self.pad,
                 len(self.payload))
 
     def save(self, path):
-        with open(path, 'wb') as f:
-            f.write(self.payload)
+        if self.hex_base_addr:
```

Comment:
This also doesn't seem so future-proof - could we have a "format" parameter to the constructor that defaults to "bin" and can be "hex" instead? That can be saved as an attribute and used here.

Reply:
Yep, that is better than what I have now